packages/ra-core/src/auth/useAuthState.ts

```diff
@@ -12,20 +12,15 @@ interface State {
 const emptyParams = {};
 
 /**
- * Hook for getting the authentication status and restricting access to authenticated users
+ * To avoid rendering a component and force waiting for the authProvider response, use the useAuthState() hook
+ * instead of the useAuthenticated() hook.
+ * It returns an object with 3 properties:
  *
- * Calls the authProvider.checkAuth() method asynchronously.
- * If the authProvider returns a rejected promise, logs the user out.
+ * - loading: true just after mount, while the authProvider is being called. false once the authProvider has answered.
+ * - loaded: the opposite of loading.
+ * - authenticated: true while loading. then true or false depending on the authProvider response.
  *
- * The return value updates according to the authProvider request state:
- *
- * - start:   { authenticated: false, loading: true, loaded: false }
- * - success: { authenticated: true,  loading: false, loaded: true }
- * - error:   { authenticated: false, loading: false, loaded: true }
- *
- * Useful in custom page components that can work both for connected and
- * anonymous users. For pages that can only work for connected users,
- * prefer the useAuthenticated() hook.
```

Comment:
I think we should keep the initial format:

> Hook for ...
> Calls the dataProvider
> State changes
> Why and when to use it

Reply:
Done. I hope is ok. Let me know